seatunnel-transforms-v2/src/main/java/org/apache/seatunnel/transform/ReplaceTransform.java

```diff
@@ -0,0 +1,107 @@
+import org.apache.seatunnel.transform.common.SeaTunnelRowAccessor;
+import org.apache.seatunnel.transform.common.SingleFieldOutputTransform;
+
+import org.apache.seatunnel.shade.com.typesafe.config.Config;
+
+import com.google.auto.service.AutoService;
+
+@AutoService(SeaTunnelTransform.class)
+public class ReplaceTransform extends SingleFieldOutputTransform {
+
+    private static final String KEY_REPLACE_FIELD = "replace_field";
+    private static final String KEY_PATTERN = "pattern";
+    private static final String KEY_REPLACEMENT = "replacement";
+    private static final String KEY_IS_REGEX = "is_regex";
+    private static final String KEY_REPLACE_FIRST = "replace_first";
+
+    private int inputFieldIndex;
+    private String replaceField;
+    private String pattern;
+    private String replacement;
+    private boolean isRegex;
+    private boolean replaceFirst;
+
+    @Override
+    public String getPluginName() {
+        return "Replace";
+    }
+
+    @Override
+    protected void setConfig(Config pluginConfig) {
+        if (!pluginConfig.hasPath(KEY_REPLACE_FIELD)) {
+            throw new IllegalArgumentException("The configuration missing key: " + KEY_REPLACE_FIELD);
+        }
+        replaceField = pluginConfig.getString(KEY_REPLACE_FIELD);
+
+        if (!pluginConfig.hasPath(KEY_PATTERN)) {
+            throw new IllegalArgumentException("The configuration missing key: " + KEY_PATTERN);
+        }
+        pattern = pluginConfig.getString(KEY_PATTERN);
+
+        if (!pluginConfig.hasPath(KEY_REPLACEMENT)) {
+            throw new IllegalArgumentException("The configuration missing key: " + KEY_REPLACEMENT);
+        }
+        replacement = pluginConfig.getString(KEY_REPLACEMENT);
+
+        if (pluginConfig.hasPath(KEY_IS_REGEX)) {
+            isRegex = pluginConfig.getBoolean(KEY_IS_REGEX);
+        }
+        if (pluginConfig.hasPath(KEY_REPLACE_FIRST)) {
+            replaceFirst = pluginConfig.getBoolean(KEY_REPLACE_FIRST);
+        }
+    }
+
+    @Override
+    protected void setInputRowType(SeaTunnelRowType rowType) {
+        inputFieldIndex = rowType.indexOf(replaceField);
+    }
+
+    @Override
+    protected String getOutputFieldName() {
+        return replaceField;
+    }
+
+    @Override
+    protected SeaTunnelDataType getOutputFieldDataType() {
+        return BasicType.STRING_TYPE;
+    }
+
+    @Override
+    protected Object getOutputFieldValue(SeaTunnelRowAccessor inputRow) {
+        Object inputFieldValue = inputRow.getField(inputFieldIndex);
+        if (inputFieldValue == null) {
+            return null;
+        }
+
+        if (isRegex) {
+            if (replaceFirst) {
+                return inputFieldValue.toString().replaceFirst(pattern, replacement);
+            }
+            return inputFieldValue.toString().replaceAll(pattern, replacement);
+        }
+        return inputFieldValue.toString().replace(pattern, replacement);
```

Comment:
If there are multiple field names containing pattern, do we need to implement a method to replace the first pattern?